src/main/java/com/google/sps/Utility.java

```diff
@@ -51,15 +52,59 @@ public static GraphNode protoNodeToGraphNode(Node thisNode) {
    * edges of the graph.
    *
    * @param graph the graph to convert into a JSON String
-   * @return a JSON object containing as entries the nodes and edges of this graph
+   * @param maxMutations the length of the list of mutations
+   * @return a JSON object containing as entries the nodes and edges of this graph as well as the
+   *     length of the list of mutations this graph is an intermediate result of applying
    */
-  public static String graphToJson(MutableGraph<GraphNode> graph) {
+  public static String graphToJson(MutableGraph<GraphNode> graph, int maxMutations) {
     Type typeOfNode = new TypeToken<Set<GraphNode>>() {}.getType();
     Type typeOfEdge = new TypeToken<Set<EndpointPair<GraphNode>>>() {}.getType();
     Gson gson = new Gson();
     String nodeJson = gson.toJson(graph.nodes(), typeOfNode);
     String edgeJson = gson.toJson(graph.edges(), typeOfEdge);
-    String resultJson = new JSONObject().put("nodes", nodeJson).put("edges", edgeJson).toString();
+    String resultJson =
+        new JSONObject()
+            .put("nodes", nodeJson)
+            .put("edges", edgeJson)
+            .put("numMutations", maxMutations)
+            .toString();
     return resultJson;
   }
+
+  /**
+   * @param original the original graph
+   * @param curr the current (most recently-requested) graph
+   * @param mutationNum number of mutations to apply
+   * @param mutList mutation list
+   * @return the resulting data graph or null if there was an error Requires that original != curr
```

Comment:
The comment "Requires that original != curr" seems to belong somewhere else. Also consider asserting the precondition is met by using the Preconditions class https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html